src/client.cc

```diff
@@ -354,9 +354,15 @@ void Client::check_if_buffer_needs_reloading()
         return;
 
     const String& filename = buffer.name();
-    timespec ts = get_fs_timestamp(filename);
-    if (ts == InvalidTime or ts == buffer.fs_timestamp())
+    const timespec ts = get_fs_timestamp(filename);
+
+    if (ts == InvalidTime)
+        return;
+    else if (MappedFile fd{filename};
+            fd.st.st_size == status.size and
+            hash_data(fd.data, fd.st.st_size) == buffer.fs_status().hash)
```

Comment:
This does not check timetamp anymore, if it did not change there is no need to map the file. And I still would prefer for `else` not to be there as we return in the previous check.